proto/vtctldata.proto

```diff
@@ -1303,6 +1359,68 @@ message ValidateVSchemaResponse {
   map<string, ValidateShardResponse> results_by_shard = 2;
 }
 
+message WorkflowDeleteRequest {
+  string keyspace = 1;
+  string workflow = 2;
+  bool keep_data = 3;
+  bool keep_routing_rules = 4;
+}
+
+message WorkflowDeleteResponse {
+  message TabletInfo {
+    string tablet = 1;
+    // The value is based on the query result's RowsAffected
+    // being 0 or not.
+    bool deleted = 2;
+  }
+  string summary = 1;
+  repeated TabletInfo details = 2;
+}
+
+message WorkflowStatusRequest {
+  string keyspace = 1;
+  string workflow = 2;
+}
+
+message WorkflowStatusResponse {
+  message TableCopyState {
+    int64 rows_copied = 1;
+    int64 rows_total = 2;
+    int32 rows_percentage = 3;
+    int64 bytes_copied = 4;
+    int64 bytes_total = 5;
+    int32 bytes_percentage = 6;
+  }
+  message ShardStreamState {
+    int32 id = 1;
+    string tablet = 2;
+    string source_shard = 3;
+    string position = 4;
+    string status = 5;
+    string info = 6;
+  }
+  message ShardStreams {
+    repeated ShardStreamState streams = 2;
+  }
+  map<string, TableCopyState> table_copy_state = 1;
+  map<string, ShardStreams> shard_streams = 2;
+}
+
+message WorkflowSwitchTrafficRequest {
+  string keyspace = 1;
+  string workflow = 2;
+  repeated string tablet_types = 3;
```

Comment:
`topodata.TabletType`?

Reply:
I thought about it. But it feels a bit silly to go from string -> TabletType -> string (that we store in the DB).  I was conflicted about it too, but it just felt like work for nothing in the end (since all we want is input validation). But I might end up doing it anyway...

Reply:
one point in favor of doing it now is it's much more annoying to do it later :) (the reverse is also true, but i am partial to structured data wherever possible, even when sometimes that data ends up as "strings that just happen to have a particular format" in the database)